### PR TITLE
Fix calls to deprecated Logger.warn/2

### PR DESCRIPTION
**Pull request by solar05**

### CODE REVIEW

**Comment by whatyouhide** [comment]
We either need to bump the version requirement to Elixir 1.11+ to do this, or check conditionally if `Logger.warning/2` is there and do that. @josevalim no preference for me.

**Comment by josevalim** [comment]
Bumping the version and CI is fine, can you please update @solar05?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -15,15 +15,15 @@ jobs:
...
-              otp: 22.3.4
           - pair:
               elixir: 1.11.4
               otp: 23.2.7
           - pair:
               elixir: 1.13.3
               otp: 24.2.2
+          - pair:
+              elixir: 1.14.5
+              otp: 25.3.2
```
Comment (whatyouhide) [inline .github/workflows/ci.yml:26]: I think we should go with the latest ones here.

```suggestion
          - pair:
              elixir: 1.15.0
              otp: 26.0.1
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -15,15 +15,15 @@ jobs:
...
-              otp: 22.3.4
           - pair:
               elixir: 1.11.4
               otp: 23.2.7
           - pair:
               elixir: 1.13.3
               otp: 24.2.2
+          - pair:
+              elixir: 1.15.0
+              otp: 26.0.1
```
Comment (josevalim) [inline .github/workflows/ci.yml:26]:
```suggestion
              elixir: 1.14.5
              otp: 25.3.2
```

File: .github/workflows/ci.yml (hunk 1 above)
Comment (josevalim) [inline .github/workflows/ci.yml:26]:
```suggestion
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Comment by StanAnsems** [comment]
This change is merged but not released as a version, is that something that will be done?

**Comment by whatyouhide** [comment]
@josevalim released 2.6.2 an hour-ish ago 🙃

**Pull request merged**